devtools/conda-envs/test_env.yaml

```diff
@@ -40,3 +41,5 @@ dependencies:
   - taproom
   - dataclasses
   - smirnoff-plugins # For testing custom nonbonded forces
+  - pip:
+    - git+https://github.com/openforcefield/openff-units.git@openmm-dalton
```

Comment:
This is https://github.com/openforcefield/openff-units/pull/11; my plan is as follows:

1. Get this passing and provisionally approved
2. Merge that PR
3. Make a new release of `openff-units`
4. Update pins here as needed
5. Aim to merge this PR